pkg/transport/graphql/response_test.go

```diff
@@ -0,0 +1,382 @@
+package graphql
+
+import (
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/jexia/semaphore/pkg/references"
+	"github.com/jexia/semaphore/pkg/specs"
+	"github.com/jexia/semaphore/pkg/specs/types"
+)
+
+func TestResponseObjectNil(t *testing.T) {
+	ResponseObject(nil, nil)
+}
+
+func TestResponseObjectInvalidType(t *testing.T) {
+	store := references.NewReferenceStore(0)
+	property := &specs.Property{
+		Template: specs.Template{
+			Scalar: &specs.Scalar{},
+		},
+	}
+
+	_, err := ResponseObject(property, store)
+	if err != ErrInvalidObject {
+		t.Fatalf("unexpected err %s, expected %s", err, ErrInvalidObject)
+	}
+}
+
+func TestResponseObject(t *testing.T) {
+	type populate func(t *testing.T, store references.Store)
+	type test struct {
+		property *specs.Property
+		populate populate
+		expected map[string]interface{}
+	}
+
+	tests := map[string]test{
```

Comment:
I would suggest moving these  definitions to an external package since all the test cases/schemas we use are pretty similar

Reply:
Agreed, I would like to create a separate PR for this to create a modular "mocking" system that has the ability to construct various structures with and without references, default types, etc.